### PR TITLE
feat(tasks): use FormBuilder to create and edit tasks.

### PR DESCRIPTION
**Pull request by pedrobonamin**
### Description

Necessary PRs:
- [x] https://github.com/sanity-io/sanity/pull/5898
- [x] https://github.com/sanity-io/sanity/pull/5900
- [x] https://github.com/sanity-io/sanity/pull/5944

#### Not `tasks` specific.
- Updates the `AddonDatasetProvider` to export a `ready` prop, which indicates that the initial load has ended.
- Adds `data-ui=fieldHeaderContentBox` into [FormFieldBaseHeader](https://github.com/sanity-io/sanity/pull/5897/files#diff-8267ccab9cea4deaa06cdb29b6e360604a499102d34e34f3272a01e45f2a4fac) to allow selecting that element and update styling accordingly. 

#### `Tasks` specific
- Introduces `FormBuilder` into tasks, by inserting a new AddonWorkspaceProvider and replicating the form builder logic which currently lives only in the `PaneProvider`.
- Adds support for `drafts`, `duplicate` tasks.
- Updates `TasksNavigation` to support `copy link` action.
- Adds comments.
- Adds activity log.

Introduces two different form for tasks, which are handled by the [`tasksSchema`](https://github.com/sanity-io/sanity/pull/5897/files#diff-a9f153e013a182d83169ce01475b060c3a67408fdb643475df06a8b6572d2a64).

#### Create form
Will be used when creating a new one, duplicating or editing a draft. 
Screenshot 2024-03-11 at 09 35 14

#### Edit form

Will be used when editing a task, this form has support for `comments` and `activityLog` 
Screenshot 2024-03-11 at 09 34 49

### What to review

Overall functionality.
Is there any harm in importing components from core into the plugin when they are not exported from `core`? See [date input
](https://github.com/sanity-io/sanity/pull/5897/files#diff-fa982587bfd087e8d9d477656d4b6796b5cc277d432671bd62249050c7db24f9R17-R19)

### Notes for release

Introduces drafts, duplicate, updated navigation and new form for tasks. Adds supports for comments and activity log in tasks.

### CODE REVIEW

File: packages/sanity/src/tasks/src/tasks/components/form/RemoveTaskDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            text: 'Remove',
+            tone: 'critical',
+            onClick: handleRemove,
+            loading: removeStatus === 'loading',
+          },
+        }}
+      >
+        <Stack space={3}>
+          <Text as="p">Are you sure you want to remove this task?</Text>
+          <Text as="p">It can't be restored</Text>
```
Comment (ninaandal) [inline packages/sanity/src/tasks/src/tasks/components/form/RemoveTaskDialog.tsx:29]: This text feels a bit off to me, suggest something like _Are you sure you want to delete this task? Once deleted, it cannot be restored._

File: packages/sanity/src/tasks/src/tasks/components/form/fields/DescriptionInput.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  return (
+    <DescriptionInputRoot $mode={mode} ref={setRootRef} $minHeight={textBoxScrollHeight || 200}>
+      <CommentInput
+        expandOnFocus={false}
+        currentUser={currentUser}
+        mentionOptions={mentionOptions}
+        onChange={handleChange}
+        value={value ?? []}
+        withAvatar={false}
+        placeholder="Optional additional description"
```
Comment (ninaandal) [inline packages/sanity/src/tasks/src/tasks/components/form/fields/DescriptionInput.tsx:80]: I know this is a design decision, but _additional_ seems unnecessary here. Suggest just _Optional description_

File: packages/sanity/src/tasks/src/tasks/components/form/tasksFormBuilder/FormCreate.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  const {onChange} = props
+  const value = props.value as TaskDocument
+  const onRemove = useCallback(() => {
+    setViewMode({type: 'list'})
+  }, [setViewMode])
+  const {handleRemove, removeStatus} = useRemoveTask({id: value._id, onRemoved: onRemove})
+
+  const handleCreate = useCallback(() => {
+    if (!value?.title) {
+      toast.push({
```
Comment (ninaandal) [inline packages/sanity/src/tasks/src/tasks/components/form/tasksFormBuilder/FormCreate.tsx:45]: This toast appears **behind** the task sidebar now, probably after Herman's changes that just got merged

File: packages/sanity/src/tasks/src/tasks/components/form/tasksFormBuilder/FormCreate.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      set(new Date().toISOString(), ['createdByUser']),
+    ])
+
+    if (createMore) {
+      setViewMode({type: 'create'})
+    } else {
+      setActiveTab('subscribed')
+    }
+
+    toast.push({
```
Comment (ninaandal) [inline packages/sanity/src/tasks/src/tasks/components/form/tasksFormBuilder/FormCreate.tsx:63]: same with this - should be on top of the sidebar

**Review by ninaandal — CHANGES_REQUESTED** [review]

**Comment by ninaandal** [comment]
All the toasts should be in front

File: packages/sanity/src/tasks/src/tasks/components/form/RemoveTaskDialog.tsx (hunk 1 above)
Comment (pedrobonamin) [reply packages/sanity/src/tasks/src/tasks/components/form/RemoveTaskDialog.tsx:29]: Makes sense! Thanks!

File: packages/sanity/src/tasks/src/tasks/components/form/fields/DescriptionInput.tsx (hunk 2 above)
Comment (pedrobonamin) [reply packages/sanity/src/tasks/src/tasks/components/form/fields/DescriptionInput.tsx:80]: I think we may want to discuss this one with Kayla next week when she is back.

File: packages/sanity/src/tasks/src/tasks/components/form/tasksFormBuilder/FormCreate.tsx (hunk 3 above)
Comment (pedrobonamin) [reply packages/sanity/src/tasks/src/tasks/components/form/tasksFormBuilder/FormCreate.tsx:45]: Seems a regression from https://github.com/sanity-io/sanity/pull/5949 
@hermanwikner could you please look at it?

File: packages/sanity/src/tasks/src/tasks/components/form/tasksFormBuilder/FormCreate.tsx (hunk 4 above)
Comment (pedrobonamin) [reply packages/sanity/src/tasks/src/tasks/components/form/tasksFormBuilder/FormCreate.tsx:63]: same as above.

File: packages/sanity/src/tasks/src/tasks/components/activity/TaskActivityEditedAt.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
+import {DotIcon} from '@sanity/icons'
```
Comment (pedrobonamin) [inline packages/sanity/src/tasks/src/tasks/components/activity/TaskActivityEditedAt.tsx:1]: This file will be updated by https://github.com/sanity-io/sanity/tree/feat/edx-1137 as a fast follow after this is merged to next

File: packages/sanity/src/tasks/src/tasks/components/activity/TasksActivityLog.tsx [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+  )
+
+  const handleCommentEdit = useCallback(
+    (id: string, next: CommentUpdatePayload) => {
+      operation.update(id, next)
+    },
+    [operation],
+  )
+
+  // TODO: Get the task real activity.
```
Comment (pedrobonamin) [inline packages/sanity/src/tasks/src/tasks/components/activity/TasksActivityLog.tsx:151]: This is coming as part of https://github.com/sanity-io/sanity/tree/feat/edx-1137

**Review by cngonzalez — APPROVED** [review]
Changes in core and structure look fine to me! I don't have enough context for the main changes, but assuming the rest of the team will look after you there. 

I'll come back with the one follow-up item I noted.

File: packages/sanity/src/tasks/src/tasks/components/form/fields/DateEditFormField.tsx [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+  type FormPatch,
+  type PatchEvent,
+  type Path,
+  set,
+  unset,
+  useDateTimeFormat,
+  useTranslation,
+} from 'sanity'
+
+// TODO: Is there a better way than importing from core like this?
```
[the remark this answers is not in the document]
Comment (pedrobonamin) [reply packages/sanity/src/tasks/src/tasks/components/form/fields/DateEditFormField.tsx:16]: Exactly, I needed a different way of using it that is not supported in the DataTimeInput. 

https://github.com/sanity-io/sanity/assets/46196328/3754cc50-4d88-4d84-b311-3128c814b7f5

**Review by ninaandal — APPROVED** [review]
🥳

**Pull request merged**